torchvision/models/detection/image_list.py

```diff
@@ -12,14 +15,61 @@ class ImageList(object):
     and storing in a field the original sizes of each image
     """
 
-    def __init__(self, tensors, image_sizes):
+    def __init__(self, tensors):
         """
         Arguments:
             tensors (tensor)
             image_sizes (list[tuple[int, int]])
         """
-        self.tensors = tensors
-        self.image_sizes = image_sizes
+        if isinstance(tensors, list):
+            print("ImageList based on a list")
+            tensors = self._batch_images(tensors).unbind()
```

Comment:
A key thing to notice here is that we're still padding the Tensors before creating the NestedTensor. A big next step will be to remove this. The assumption that each entry is of the same shape is deeply ingrained into the codebase.